src/plugins/intel_gpu/src/kernel_selector/kernels/permute/permute_kernel_f_y_axes.cpp

```diff
@@ -0,0 +1,234 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "permute_kernel_f_y_axes.h"
+
+#include <algorithm>
+#include <cmath>
+#include <string>
+
+#include "common_tools.h"
+#include "kernel_selector_utils.h"
+
+namespace kernel_selector {
+
+namespace {
+
+constexpr size_t cSimpleMemCopyOpDivider = 4UL;
+constexpr size_t c3DTransposeBufHeight = 4UL;
+
+size_t GetDivisor(const size_t input_size) {
+    std::vector<size_t> v = {/*32,*/ 16, 8, 4, /*2,*/ 1};
+    auto is_divided = [input_size](size_t i) {
+        return input_size % i == 0;
+    };
+    auto result = std::find_if(begin(v), end(v), is_divided);
+    return *result;
+}
+
+bool IsSimpleMemCopyOperation(const permute_params& params) {
+    return params.inputs[0].X().v > 1 && params.inputs[0].GetLayout() == DataLayout::bfyx;
+}
+
+bool Is3DTranspose(const permute_params& params) {
+    return params.inputs[0].X().v > 1 && params.inputs[0].GetLayout() != DataLayout::bfyx;
+}
+
+size_t GetFeatureBlockSize(const permute_params& params) {
+    return std::min(GetDivisor(params.inputs[0].Feature().v), GetDivisor(params.inputs[0].Y().v));
+}
+
+size_t GetTileHeight(const permute_params& params) {
```

Comment:
Do we need to maintain tile width/height separately? I think it make things too compilcated. Previously when we implemented permute_kernel_tile_4x4_8x8, we diddnt see that much gain with independent tile width/height. Things had better being simplified unless it is proven by an extensive perf evaluation.